AddinFiles/main.jsl

```diff
@@ -484,12 +495,820 @@ BuildJSL = function( {}, {txt},
+			if(!contains({"Point Info", "Snapshot Values"}, expr(querytype)),
+				dtCount = dtLong << summary(group(:tag), "private");
+				rowCount = colMax(column(dtCount, "N Rows"));
+				close(dtCount, "nosave");
+			,
+				rowCount = nRows(dtLong)
+			);
+			if(rowCount >= ns:maxRows,
+				NewWindow("Max Rows Exceeded", <<modal,
+					TextBox(
+						"Met or exceeded the maximum number of rows (" || char(ns:maxRows) ||
+						") for this query. Increase the allowed maxrows or restrict the query. "||
+						" Some results might be missing."
+					)
+				);
+				dtLong << BringWindowToFront;
+			);
+			oldcn = expr(oldcn);
+			oldcncorrected = oldcn;
+		));
+
+		If(cnformat != "tag",
+			//Change Column Names 
+			InsertInto(NewTblExpr, EvalExpr(
+				newcn = expr(newcn);
+				For( i = 1, i <= N Rows( dtLong ), i++,
+					pos = Min( Loc( oldcncorrected, Uppercase( Column( dtLong, "tag" )[i]) ) );
+					If( pos > 0,
+						Column( dtLong, "tag" )[i] = newcn[pos]
+					)
+				)
+			))
+		);
+
+		If( TableFormat == "Wide",
+			// Make wide table
+			InsertInto(NewTblExpr, EvalExpr(
+				tableName = dtLong << getName;
+				dtWide = dtLong << Split(
+					Split By( :tag ),
+					Split( expr(ColToSplit) ),
+					Group( :time ),
+					Remaining Columns( Drop All ),
+					Sort by Column Property
+				);
+				close(dtLong, "nosave");
+				dtWide << setName(tableName);
+			));
+			If(cb_tagtonotes << Get,
+				// Save column properties
+				InsertInto(NewTblExpr, EvalExpr(
+					For( i = 1, i <= N Cols( dtWide ), i++,
+						cn = Regex( Column( dtWide, i ) << Get Name, "((value )|(svalue )|(status ))?(.*)", "\5" );
+						If( !Is Missing( cn ),
+							pos = Min( Loc( oldcn, cn ) );
+							If( pos > 0,
+								Column( dtWide, i ) << Set Property( "PI tag", oldcncorrected[pos] );
+								Column( dtWide, i ) << Set Property( "PI Original Column Name", Column( dtWide, i ) << Get Name );
+								Column( dtWide, i ) << Set Property( "PI call type", expr(querytype) );
+								Column( dtWide, i ) << Set Property( "PI interval", expr(intervalTime) );
+								Column( dtWide, i ) << Set Property( "PI Server", expr(serverName) );
+								If( contains({"Average", "Count", "Max", "Min", "Population Standard Deviation", "Range", "Standard Deviation", "Totals"}, expr(querytype)), 
+									Column( dtWide, i ) << Set Property( "PI step", expr(stepTime));
+									Column( dtWide, i ) << Set Property( "PI interval start offset", "-" || expr(stepTime));
+									Column( dtWide, i ) << Set Property( "PI interval end offset", "");
+								)
+							);
+						);
+					)
+				))
+			)
+		);
+		InsertInto( NewTblExpr , Eval Expr(
+			dt = new Table("test");
+			dt = DataTable(tableName);
+			dt;
+		));
+		Return( NameExpr(NewTblExpr) );
+	);
+);
 //flag to output a bunch of status info to the log file
 debug = 0;
 
 //window box, all code is inside this box
-nw = New Window( "PI Web API Query Builder",
+nw = New Window( "PI Web API Query Builder(JMP Desktop Only)",
```

Comment:
Does the JMP Desktop Only text need to stay here?

Reply:
I think so